certbot-apache/certbot_apache/interfaces.py

```diff
@@ -242,7 +208,7 @@ def set_parameters(self, parameters):
 
 
 @six.add_metaclass(abc.ABCMeta)
-class BlockNode(ParserNode):
+class BlockNode(object):
```

Comment:
Why remove the inheritance from `ParserNode`?

Reply:
I thought the suggestion for using ABC `register()` functionaity was aimed towards the interfaces as well. I hit some linter errors about not implementing the superclass methods and thought that if we'd be using the register() regardless, it would not matter.

I can return the inheritance and add the linter exceptions there as well though.

Reply:
My suggestion to use `register` [here](https://github.com/certbot/certbot/pull/7308#discussion_r313650641) was to define an `AugeasParserNode` _implementation_ so that we weren't duplicating the same `ParserNode` code 3 times and use it without resorting to multiple inheritance. This looks something like:
```
class _AugeasParserNode(interfaces.ParserNode):
    def save():
    ....
class AugeasCommentNode(_AugeasParserNode):
    def __init__():
    ....
AugeasCommentNode.register(interfaces.CommentNode)
```
You could also probably have `_AugeasParserNode` inherit from `object` and register it if you prefer.

As for the interfaces modified in this PR, the three real node interfaces are now defined in a way where to implement the class you do not need to defined the `ParserNode` methods which seems like a problem that should be fixed to me instead of making implementations explicitly register themselves with multiple interfaces. I don't think we ever want a `CommentNode` that is not a `ParserNode`.

I think continuing to use inheritance between the interfaces and fixing the `pylint` issue which is a [known bug in `pylint`](https://github.com/PyCQA/pylint/issues/179) is the best approach.

Reply:
My suggestion to use `register` [here](https://github.com/certbot/certbot/pull/7308#discussion_r313650641) was to define an `AugeasParserNode` _implementation_ so that we weren't duplicating the same `ParserNode` code 3 times and use it without resorting to multiple inheritance. This looks something like:
```
class _AugeasParserNode(interfaces.ParserNode):
    def save():
    ....
class AugeasCommentNode(_AugeasParserNode):
    def __init__():
    ....
AugeasCommentNode.register(interfaces.CommentNode)
```
You could also probably have `_AugeasParserNode` inherit from `object` and register it if you prefer.

As for the interfaces modified in this PR, the three real node interfaces are now defined in a way where to implement the class you do not need to defined the `ParserNode` methods which seems like a problem that should be fixed to me instead of making implementations explicitly register themselves with multiple interfaces. I don't think we ever want a `CommentNode` that is not a `ParserNode`.

I think continuing to use inheritance between the interfaces and fixing the `pylint` issue which is a [known bug in `pylint`](https://github.com/PyCQA/pylint/issues/179) is the best approach.

Reply:
Thanks for the pointer! I was getting pretty frustrated there. Unfortunately only a file-wide pylint exclude worked here for a reason or another. It's now fixed however.